lib/srv/sess.go

```diff
@@ -686,11 +484,19 @@ func newSession(id rsession.ID, r *SessionRegistry, ctx *ServerContext) (*sessio
 		presenceEnabled:                ctx.Identity.Certificate.Extensions[teleport.CertExtensionMFAVerified] != "",
 		io:                             NewTermManager(),
 		stateUpdate:                    sync.NewCond(&sync.Mutex{}),
-		doneCh:                         make(chan struct{}, 2),
+		doneCh:                         make(chan struct{}),
 		initiator:                      ctx.Identity.TeleportUser,
 		displayParticipantRequirements: utils.AsBool(ctx.env[teleport.EnvSSHSessionDisplayParticipantRequirements]),
 	}
 
+	sess.io.OnWriteError = func(idString string, err error) {
+		if idString == sessionRecorderID {
+			sess.log.Error("Failed to write to session recorder, stopping session.")
+			// stop in goroutine to avoid deadlock
+			go sess.Stop()
```

Comment:
This stops the session if any party has a flaky connection, would it be better to force the party to leave and remove the writer? If the party was required for the session to run as per moderated sessions, the leave mechanism should kick in and stop it.

Reply:
Nevermind, I misread this